servers/visual/visual_server_scene.h

```diff
@@ -355,6 +355,19 @@ class VisualServerScene {
 						visible == p_cache.visible);
 			}
 
+			bool operator!=(const LightCache &p_cache) {
+
+				return (type != p_cache.type ||
```

Comment:
I think, to avoid mistakes, the operator definition should be like this:

bool operator!=(const LightCache &p_cache) {
    return !operator==(p_cache);
}